src/client/shadows/dynamicshadowsrender.cpp

```diff
@@ -212,22 +243,60 @@ void ShadowRenderer::update(video::ITexture *outputTarget)
 			// Depth texture is available in irrlicth maybe we
 			// should put some gl* fn here
 
-			if (light.should_update_map_shadow) {
-				light.should_update_map_shadow = false;
 
-				m_driver->setRenderTarget(shadowMapClientMap, true, true,
+			if (m_current_section < total_sections) {
+				m_driver->setRenderTarget(shadowMapClientMapFuture, reset_sm_texture, true,
 						video::SColor(255, 255, 255, 255));
-				renderShadowMap(shadowMapClientMap, light);
+				renderShadowMap(shadowMapClientMapFuture, light, m_current_section, total_sections);
 
 				if (m_shadow_map_colored) {
-					m_driver->setRenderTarget(shadowMapTextureColors,
-							true, false, video::SColor(255, 255, 255, 255));
+					m_driver->setRenderTarget(shadowMapTextureColorsFuture,
+							reset_sm_texture, false, video::SColor(255, 255, 255, 255));
 				}
-				renderShadowMap(shadowMapTextureColors, light,
+				renderShadowMap(shadowMapTextureColorsFuture, light, m_current_section, total_sections,
 						scene::ESNRP_TRANSPARENT);
 				m_driver->setRenderTarget(0, false, false);
 			}
 
+			reset_sm_texture = false;
+		} // end for lights
+
+		// move to the next section
+		++m_current_section;
+
+		// pass finished, swap textures and commit light changes
+		if (m_current_section == total_sections) {
+			auto tmp = shadowMapClientMapFuture;
+			shadowMapClientMapFuture = shadowMapClientMap;
+			shadowMapClientMap = tmp;
+
+			tmp = shadowMapTextureColorsFuture;
+			shadowMapTextureColorsFuture = shadowMapTextureColors;
+			shadowMapTextureColors = tmp;
```

Comment:
`std::swap`

Reply:
It can even be `std::move` as `tmp` is not reutilized afterward.